### ISSUE TITLE
ScriptParameterParseException when upgrading from 1.4 -> 1.5.2 

### ISSUE THREAD

**Issue by aewhite** [elastic/elasticsearch#10926]
After upgrading from 1.4 -> 1.5.2 shards fail to load due to ScriptParameterParseException. 

Stacktrace:

```
org.elasticsearch.script.ScriptParameterParser$ScriptParameterParseException: Value must be of type String: [lang]
        at org.elasticsearch.script.ScriptParameterParser.parseConfig(ScriptParameterParser.java:111)
        at org.elasticsearch.index.mapper.DocumentMapperParser.parseTransform(DocumentMapperParser.java:307)
        at org.elasticsearch.index.mapper.DocumentMapperParser.parse(DocumentMapperParser.java:257)
        at org.elasticsearch.index.mapper.DocumentMapperParser.parseCompressed(DocumentMapperParser.java:192)
        at org.elasticsearch.index.mapper.MapperService.parse(MapperService.java:434)
        at org.elasticsearch.index.mapper.MapperService.merge(MapperService.java:307)
        at org.elasticsearch.indices.cluster.IndicesClusterStateService.processMapping(IndicesClusterStateService.java:430)
        at org.elasticsearch.indices.cluster.IndicesClusterStateService.applyMappings(IndicesClusterStateService.java:376)
        at org.elasticsearch.indices.cluster.IndicesClusterStateService.clusterChanged(IndicesClusterStateService.java:181)
        at org.elasticsearch.cluster.service.InternalClusterService$UpdateTask.run(InternalClusterService.java:467)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.runAndClean(PrioritizedEsThreadPoolExecutor.java:188)
        at org.elasticsearch.common.util.concurrent.PrioritizedEsThreadPoolExecutor$TieBreakingPrioritizedRunnable.run(PrioritizedEsThreadPoolExecutor.java:158)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1145)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:615)
        at java.lang.Thread.run(Thread.java:745)
```

The cause as I can best understand is the `lang` property is getting set to null for some transform scripts. We do not explicitly set the lang so this was unexpected. Our process is to use a template...

```
"mappings": {
            "<type>": {                
                "transform": [
                    {"script": "<source>"},
...
```

However, the mapping produced after an index creation (in 1.4) is...

```
{
    "<index-name>": {
        "mappings": {
            "<type>": {
                "transform": [
                    {"script": "<source>","lang": null}
...
```

The above mapping works in 1.4 but not in 1.5 it seems.

**Comment by clintongormley** [comment]
@colings86 please could you take a look.  probably related to #7977

**Comment by aewhite** [comment]
Note to others; the only workaround I have found that works is to recreate the index and set lang property on transform explicitly. Perhaps there is a way to update an existing transform but it is not mentioned in the [Transform Documentation](http://www.elastic.co/guide/en/elasticsearch/reference/1.x/mapping-transform.html) at this time.

**Comment by colings86** [comment]
@aewhite thanks for raising this. When the transform is written in the mapping the script language is written even if it is set to null (null meaning use the default). I have opened PR #10976 which lets the parser reading the stored mapping read null for the script language instead of throwing an error

**Comment by colings86** [comment]
@aewhite thanks again for raising this. The fix has now been merged in and should be available from 1.5.3
